### PR TITLE
Show glycemia by date

### PR DESCRIPTION
**Pull request by M0UTOO**

### CODE REVIEW

File: entries/entry.c [hunk 1]
```diff
@@ -215,3 +216,110 @@ void showEntries(Entry *glycemia) {
...
+         printf("| Comment    : %s\n",glycemia->comment);
+         printf(" ------------\n\n");
+
+      }
+      glycemia = glycemia->next;
+   }
+
+}
+
+void showEntriesBeforeAfterDate(Entry *glycemia, char *date, int user_id, sqlite3 *db, char *zErrMsg, int rc, bool before) {
```
Comment (Mil0w0) [inline entries/entry.c]: no need of Entry glycemia in this and the following functions.

File: main.c [hunk 2]
```diff
@@ -254,6 +253,52 @@ int main(int argc, char **argv)
...
+            printf("4. See your glycemia between two dates\n"); 
+            printf("0. Exit Logs from Date\n");
+            scanf(" %c", &choice);
+            
+            if(choice == '1')
+            {
+                printf("\nPlease enter the date you want to see the logs for (dd/mm/yyyy):\n");
+                scanf("%s", &date);
+                //show glycemia logs for a specific date
+                showEntriesForDate(glycemia, date);
```
Comment (Mil0w0) [inline main.c:272]: Need to check if the user input isn't dangerous and is of correct format.
Risk of SQL Injection ?

File: main.c [hunk 3]
```diff
@@ -22,6 +22,8 @@ int main(int argc, char **argv)
     int connected = 0;
     char username[30];
     char password[30];
+    char date[20];
```
Comment (Mil0w0) [inline main.c]: why 20 ? make it the exact size of the date format if no reasons

**Review by Mil0w0 — CHANGES_REQUESTED** [review]
Overall it's a great job ! Minor changes required and then it will be all good

File: main.c (hunk 2 above)
Comment (M0UTOO) [reply main.c:272]: no need cause of the size of date :)

File: entries/entry.c (hunk 1 above)
Comment (M0UTOO) [reply entries/entry.c]: Done !

File: main.c (hunk 3 above)
Comment (M0UTOO) [reply main.c]: Size is now correct

**Review by Mil0w0 — APPROVED** [review]
Fine

**Pull request merged**